src/installer/installer.go

```diff
@@ -253,6 +318,40 @@ func (i *installer) writeImageToDisk(ignitionPath string) error {
 		i.log.Errorf("Failed to write image to disk %s", err)
 		return err
 	}
+	return nil
+}
+
+func (i *installer) writeReleaseOsImageToDisk(ignitionPath string) error {
+
+	var mc *mcfgv1.MachineConfig
+	egroup, ctx := errgroup.WithContext(context.Background())
+
+	egroup.Go(i.getEncapsulatedMC(ctx, ignitionPath, &mc))
+	egroup.Go(i.writeImageForOverwrite(ctx, ignitionPath))
+
+	err := egroup.Wait()
+	if err != nil {
+		return err
+	}
+	if err = i.overwriteOsImage(mc.Spec.OSImageURL, mc.Spec.KernelArguments); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *installer) writeImageToDisk(ignitionPath string) error {
+	i.UpdateHostInstallProgress(models.HostStageWritingImageToDisk, "")
+
+	var err error
+	if i.EnableSkipMcoReboot {
+		err = i.writeReleaseOsImageToDisk(ignitionPath)
```

Comment:
In case this fail let's log it and publish an event but let's not fail the installation, instead we can just fall back to  `writeRunningOsImageToDisk`

Reply:
First line in writeReleaseOsImageToDisk is writeRunningOsImageToDisk , why we can't always run writeRunningOsImageToDisk first and only then run all other logic from writeRunningOsImageToDisk ?